test/utils.py

```diff
@@ -35,6 +35,24 @@
 }.get(detect_ci_provider(), True)
 
 
+def _subprocess_run(command: list[str], **kwargs: Any) -> None:
+    result = subprocess.run(
+        command,
+        capture_output=True,
+        check=False,
+        text=True,
+        **kwargs,
+    )
+    if result.returncode != 0:
+        msg = (
+            f"Subprocess failed with exit code {result.returncode}\n"
+            f"Command: {' '.join(command)}\n"
+            f"--- stdout ---\n{result.stdout}\n"
+            f"--- stderr ---\n{result.stderr}"
+        )
+        raise AssertionError(msg)
```

Comment:
The `AssertionError` does not play well with tests expecting a `subprocess.CalledProcessError`
I'm not sure we could get a clean output with `subprocess.CalledProcessError` in which case we might want to create our own exception for this in order to get a clean output & maybe still register the return code & possibly stdout/stderr depending on wether or not they're used in tests expecting a `subprocess.CalledProcessError`.